File: Week 1/vigenere.cpp
#include<iostream>
#include<string>
#include <cctype>
using namespace std;

void encryption(string plaintext,string keyword){
    for(int i=0; i < plaintext.length(); i++){
        plaintext[i] = toupper(plaintext[i]);
    }
    for(int i=0; i < keyword.length(); i++){
        keyword[i] = toupper(keyword[i]);
    }

    string ciphertext = "";

    for (int i=0; i < plaintext.length(); i++){
        if (isalpha(plaintext[i])){
            char p_index = plaintext[i] - 'A';
            char k_index= keyword[i%keyword.length()] - 'A';
            ciphertext += (p_index + k_index)%26 + 'A';
        }
        else {
            ciphertext += plaintext[i];
        }
    }
    cout<<ciphertext;
}

void decryption(string ciphertext,string keyword){
    for(int i=0; i < ciphertext.length(); i++){
        ciphertext[i] = toupper(ciphertext[i]);
    }
    for(int i=0; i < keyword.length(); i++){
        keyword[i] = toupper(keyword[i]);
    }

    string plaintext = "";

    for (int i=0; i < ciphertext.length(); i++){
        if (isalpha(ciphertext[i])){
            char c_index = ciphertext[i] - 'A';
            char k_index= keyword[i%keyword.length()] - 'A';
            plaintext += (c_index - k_index +26 ) % 26 + 'A';
        }
        else {
            plaintext += ciphertext[i];
        }
    }
    cout<<plaintext;
}


int main(){
    string text;
    string keyword;
    char choice;
    cout << "Enter text: ";
    getline(cin, text);
    cout << "Enter shift key: ";
    getline(cin, keyword);
    cout << "What do you want to do? E/D :";
    cin>>choice;
    if(choice == 'E'){
        encryption(text,keyword);
    }
    else if(choice == 'D'){
        decryption(text,keyword);
    }
    else{
        cout<<"Enter Valid Choice !";
    }
    
   return 0; 
}

File: Week 1/vigenere.py
def encryption(plaintext,keyword):
    plaintext=plaintext.upper()
    keyword=keyword.upper()
    ciphertext = ""
    key_len=len(keyword)
    for i in range(len(plaintext)):
        if plaintext[i].isalpha():
            p_index = ord(plaintext[i])-65
            k_index= ord(keyword[i%key_len])-65
            c_index=(p_index+k_index)%26
            ciphertext += chr(c_index + 65)
        else:
            ciphertext+=plaintext[i]
    print(ciphertext)

def decryption(ciphertext,keyword):
    ciphertext=ciphertext.upper()
    keyword=keyword.upper()
    plaintext = ""
    key_len=len(keyword)
    for i in range(len(ciphertext)):
        if ciphertext[i].isalpha():
            c_index = ord(ciphertext[i])-65
            k_index= ord(keyword[i%key_len])-65
            p_index=(c_index-k_index+26)%26
            plaintext += chr(p_index + 65)
        else:
            plaintext+=ciphertext[i]
    print(plaintext)

text = input("Enter text: ")
keyword = input("Enter Keyword: ")
choice = input("What do you want to do? E/D :")

if choice=='E':
    encryption(text,keyword)
elif choice=='D':
    decryption(text,keyword)
else:
    print("Enter Valid Choice !")

